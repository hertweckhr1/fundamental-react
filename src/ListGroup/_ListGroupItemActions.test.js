/* eslint-disable jsx-a11y/anchor-is-valid */
import { Button } from '../';
import ListGroupItemActions from './_ListGroupItemActions';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<ListGroupItemActions />', () => {
    const listGroupAction = (
        <ListGroupItemActions>
            <Button glyph='edit' type='standard' />
        </ListGroupItemActions>
    );

    test('create list group item action', () => {
        // list group with action
        let component = renderer.create(listGroupAction);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the ListGroupItemActions component', () => {
            const element = mount(<ListGroupItemActions data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
