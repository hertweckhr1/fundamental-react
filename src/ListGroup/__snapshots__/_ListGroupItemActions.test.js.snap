// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGroupItemActions /> create list group item action 1`] = `
<span
  className="fd-list-group__action"
>
  <button
    className="fd-button fd-button--standard sap-icon--edit"
  />
</span>
`;
